x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/support/BaseMlIntegTestCase.java

```diff
@@ -315,6 +316,38 @@ public static void indexDocs(Logger logger, String index, long numDocs, long sta
         logger.info("Indexed [{}] documents", numDocs);
     }
 
+    public static void indexDocs(Logger logger, String index, long numDocs, long start, long end, TimeValue interval) {
```

Comment:
I am a bit worried about adding a second version of `indexDocs` here. I think the problem with the test is that we need to ensure the first bucket is not partial. Could we not change the existing `indexDocs` to always index the first record with `start` time and then change the test to do the bucket alignment?